sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -101,6 +112,46 @@ def invoke(
     can be a bag of computed values (Ts or Awaitable[T]s), and the result is a Awaitable[Any] that
     resolves when the invoke finishes.
     """
+    return _invoke(tok, props, opts, typ, run_async=False)
+
+
+async def invoke_async(
+    tok: str,
+    props: "Inputs",
+    opts: Optional[InvokeOptions] = None,
+    typ: Optional[type] = None,
+) -> Any:
```

Comment:
this is probably the right signature, but it is a little odd that this isn't returning an `InvokeResult`. returning an `InvokeResult` would allow callees of `invoke` to transition to `invoke_async` by just changing the function name and adding `await`.

Reply:
Ah, I had added some comments to the PR before I tagged you, but I guess I never clicked "submit" on those. One of the comments touched on this. Went ahead and submitted those.

In any case, yeah, I went back and forth on this. It doesn't strictly need to be `InvokeResult` -- our other SDKs like Node don't doe this, they just return `Promise<T>`. And originally the Python SDK didn't until we reworked the previous async invokes to be synchronous.

Reply:
One other thing... I don't really expect much use of this directly by users. This is a pretty advanced use case. The vast majority of users are going to be using generated provider SDK invokes. So not sure we need to optimize for avoiding having to remove a `.value` when transitioning. I almost lean towards the "right" signature in this case.

Reply:
Sounds good to me 👍